pkg/health/clusterStatsHandler.go

```diff
@@ -259,72 +259,176 @@ func ProcessClusterIngestStatsHandler(ctx *fasthttp.RequestCtx, orgId int64) {
 		httpResp.ChartStats[k]["MetricsBytesCount"] = float64(entry.MetricsBytesCount)
 		httpResp.ChartStats[k]["TraceBytesCount"] = float64(entry.TraceBytesCount)
 		httpResp.ChartStats[k]["TraceSpanCount"] = entry.TraceSpanCount
+		httpResp.ChartStats[k]["ActiveSeriesCount"] = entry.ActiveSeriesCount
 	}
 	utils.WriteJsonResponse(ctx, httpResp)
 }
 
+func parseIngestionStatsRequest(jsonSource map[string]interface{}) (uint64, usageStats.UsageStatsGranularity) {
+	defaultPastHours := uint64(7 * 24) // 7 days default
+
+	startEpoch, hasStart := jsonSource["startEpoch"]
+	endEpoch, hasEnd := jsonSource["endEpoch"]
+	granularity, hasGranularity := jsonSource["granularity"]
+
+	getDefault := func() (uint64, usageStats.UsageStatsGranularity) {
+		if hasGranularity {
+			return defaultPastHours, parseGranularity(granularity)
+		}
+		return defaultPastHours, determineGranularity(defaultPastHours)
+	}
+
+	if !hasStart || !hasEnd || startEpoch == nil || endEpoch == nil {
+		return getDefault()
```

Comment:
nit: a little hard to follow the code, you probably dont need the getDefault func enclosure, you can just write that code in here